frappe/email/doctype/email_account/email_account.json

```diff
@@ -601,6 +602,14 @@
    "fieldtype": "Select",
    "label": "Method",
    "options": "Basic\nOAuth"
+  },
+  {
+   "default": "0",
+   "depends_on": "eval:!doc.domain && doc.enable_incoming",
```

Comment:
I made a change on that

the 
!doc.domain
 is the same as use_ssl, I guess as this is a similar option it have to be rule by the same behaviour

I also add,  
doc.enable_incoming && doc.use_imap 

and !doc.use_ssl

Because STARTTLS is possible only if it's not SSL.

I aloso update doctypes validate method to clean use_starttls if use_ssl is set